sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1385 @@
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DetectLanguageResult>> detectLanguageWithResponse(String text, String countryHint) {
+        try {
+            return withContext(context -> detectLanguageWithResponse(text, countryHint, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DetectLanguageResult>> detectLanguageWithResponse(String text, String countryHint, Context context) {
+        Objects.requireNonNull(text, "'text' cannot be null.");
+        List<DetectLanguageInput> languageInputs = Collections.singletonList(new DetectLanguageInput("0",
+            text, countryHint));
+        return detectBatchLanguagesWithResponse(languageInputs, null, context).map(response ->
+            new SimpleResponse<>(response, response.getValue().iterator().next()));
+    }
+
+    /**
+     * Returns the detected language for a batch of input.
+     *
+     * @param textInputs The list of texts to be analyzed.
+     *
+     * @return A {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     *
+     * @throws NullPointerException if {@code textInputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DocumentResultCollection<DetectLanguageResult>> detectLanguages(List<String> textInputs) {
+        try {
+            return detectLanguagesWithResponse(textInputs, defaultCountryHint).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns the detected language for a batch of input with the provided country hint.
+     *
+     * @param textInputs The list of texts to be analyzed.
+     * @param countryHint A country hint for the entire batch. Accepts two letter country codes specified by ISO
+     * 3166-1 alpha-2. Defaults to "US" if not specified.
+     *
+     * @return A {@link Response} of {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     *
+     * @throws NullPointerException if {@code textInputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(
+        List<String> textInputs, String countryHint) {
+        try {
+            return withContext(context -> detectLanguagesWithResponse(textInputs, countryHint, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(List<String> textInputs,
+        String countryHint, Context context) {
+        Objects.requireNonNull(textInputs, "'textInputs' cannot be null.");
```

Comment:
No null validation required for `countryHint` and `context`?

Reply:
The context should come from the above call that calls this method and yes countryHint can be null. That is service side validation.

Reply:
country hint won't be required. For context, do we usually required? I don't see it is required else